### ISSUE TITLE
ParamSpec fails to bind to outer scope in nested function.

### ISSUE THREAD

**Issue by bijij** [microsoft/pyright#1971]
**Describe the bug**
Paramspec will 

**To Reproduce**
see code snippet

Pyright will return the following.

`test.py:17:12 - error: Expression of type "(func: Callable[P@inner, Any]) -> Foo[P@inner]" cannot be assigned to return type "(_p0: Callable[P@foo, Any]) -> Foo[P@foo]"
    Type "(func: Callable[P@inner, Any]) -> Foo[P@inner]" cannot be assigned to type "(_p0: Callable[P@foo, Any]) -> Foo[P@foo]"
      Function return type "Foo[P@inner]" is incompatible with type "Foo[P@foo]"
        TypeVar "P@Foo" is invariant
          Type "P@inner" cannot be assigned to type "P@foo" (reportGeneralTypeIssues)`

**Expected behavior**
Type checking would pass

**Screenshots or Code**
```python
from typing import Any, Callable, Generic
from typing_extensions import ParamSpec

P = ParamSpec('P')

class Foo(Generic[P]):
    
    def __init__(self, func: Callable[P, Any]) -> None:
        ...


def foo() -> Callable[[Callable[P, Any]], Foo[P]]:
    
    def inner(func: Callable[P, Any]) -> Foo[P]:
        return Foo[P](func)

    return inner

```

![image](https://user-images.githubusercontent.com/8677174/121529175-7c2ce700-ca3f-11eb-8745-d24c712bcb69.png)

**VS Code extension or command-line**
pyright 1.1.147

**Additional info**
Not sure if this is allowed, skimming through the PEP I don't see this use case mentioned, type-ignoring the return variable will acheive the desired result either way,

**Comment by erictraut** [comment]
This is an unusual use, but at first glance, I think it should be allowed. Currently, the type checker is binding the inner `P` to the function `inner` rather than using the scope defined by `foo`. Since `P@foo` and `P@inner` are treated as different, that explains the type consistency error.

I confirmed that pyright does properly handle non-ParamSpec TypeVars in this manner, so it makes sense that ParamSpecs should be handled in this case as well.

```python
T = TypeVar("T")


def bar(x: T) -> T:
    def inner(y: T) -> T: # T is bound to "bar" in this case
        return y

    return inner(x)
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.149, which I just published. It will also be in the next release of pylance.
